### PR TITLE
Flats filters

### PR DESCRIPTION
**Pull request by globaevg**

### CODE REVIEW

File: server/repositories/flat.js [hunk 1]
```diff
@@ -3,10 +3,16 @@ const FlatsReserve = require('../models/FlatsReserve');
...
+      [currentFilter]: parseInt(moreless)
+    };
+
+    const countQuery = Flat.countDocuments();
+
     const flatsQuery = Flat.find()
-      .sort({ date: -1 })
+      .sort(paramFilter)
+      .collation({ locale: 'en_US', numericOrdering: true })
       .limit(+pageSize)
```
Comment (antonstan) [inline server/repositories/flat.js]: dont cast types using + symbol. it is not readable at all. please use parseInt function

File: client/src/components/pages/flats/Flats.jsx [hunk 2]
```diff
@@ -2,35 +2,77 @@ import React, { Component, Fragment } from 'react';
...
-    getFlats(pageNumber - 1, this.state.pageSize);
+    getFlats(pageNumber - 1, pageSize, currentFilter, moreless);
     this.setState({ currentPage: pageNumber });
   };
 
+  changeFilter = ({ target }) => {
+    const { getFlats } = this.props;
+    const { currentPage, pageSize, currentFilter, moreless } = this.state;
+    if (currentFilter == target.value) {
+      this.setState({ moreless: -this.state.moreless, currentPage: 1 });
```
Comment (antonstan) [inline client/src/components/pages/flats/Flats.jsx]: -this.state.moreless you already get this var from state. no need to use it with this.state prefix - just moreless

Comment (antonstan) [reply client/src/components/pages/flats/Flats.jsx]: and what is the weird name of the variable "moreless" . what does this option should do?

Comment (globaevg) [reply client/src/components/pages/flats/Flats.jsx]: "moreless" define, how to sort list, from less to more or reverse

**Pull request merged**